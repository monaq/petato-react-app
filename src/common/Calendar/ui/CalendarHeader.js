import React from "react";
import PropTypes from 'prop-types';

const CalendarHeader = propValues => {
  const { title, onPrevClick, onNextClick, month, year } = propValues;

  return (
    <div className="c-calendar__header">
      <div
        role="button"
        className={["c-calendar-arrow", "c-calendar-arrow__prev"].join(" ")}
        onClick={onPrevClick}
      >
        ◀
      </div>
      <h1 className="c-calendar-title">{title}</h1>
      <div
        role="button"
        className={["c-calendar-arrow", "c-calendar-arrow__next"].join(" ")}
        onClick={onNextClick}
      >
        ▶
      </div>
    </div>
  );
};

CalendarHeader.propTypes = {
  onNextClick: PropTypes.func,
  onPrevClick: PropTypes.func
};

export default CalendarHeader;
