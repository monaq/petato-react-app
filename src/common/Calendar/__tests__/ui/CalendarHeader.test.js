import Adapter from "enzyme-adapter-react-16";
import CalendarHeader from "../../ui/CalendarHeader";
import commonProps from "../fixtures/common.json";
import { mount, shallow } from "enzyme";
import React from "react";

const { monthFormat, dayFormat } = commonProps;

const given = {
  title: "2019년 3월",
  onPrevClick: jest.fn(),
  onNextClick: jest.fn(),
  month: 3,
  year: 2019
};

beforeEach(() => {
  given.onPrevClick.mockClear();
  given.onNextClick.mockClear();
});

describe("CalendarHeader", () => {
  it("should render title", () => {
    const component = shallow(<CalendarHeader {...given} />);
    expect(component.html()).toMatchSnapshot();
  });

  it("should call props.onPrevClick on click prev button", () => {
    const component = shallow(<CalendarHeader {...given} />);
    component.find('.c-calendar-arrow__prev').simulate('click');
    expect(given.onPrevClick).toHaveBeenCalled();
  });
  it("should call props.onNextClick on click next button", () => {
    const component = shallow(<CalendarHeader {...given} />);
    component.find('.c-calendar-arrow__next').simulate('click');
    expect(given.onNextClick).toHaveBeenCalled();
  });
});
