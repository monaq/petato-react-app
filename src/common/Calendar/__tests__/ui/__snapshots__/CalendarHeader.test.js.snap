// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarHeader should render title 1`] = `"<div class=\\"c-calendar__header\\"><div role=\\"button\\" class=\\"c-calendar-arrow c-calendar-arrow__prev\\">◀</div><h1 class=\\"c-calendar-title\\">2019년 3월</h1><div role=\\"button\\" class=\\"c-calendar-arrow c-calendar-arrow__next\\">▶</div></div>"`;
